devicetypes/fibargroup/fibaro-walli-roller-shutter-driver.src/fibaro-walli-roller-shutter-driver.groovy

```diff
@@ -0,0 +1,549 @@
+		if (endpoint) {
+			cmd = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+		}
+		if (zwaveInfo.zw.contains("s")) {
+			zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+private getParameterMap() {[
+		[
+				name: "LED frame - colour when moving", key: "ledFrame-ColourWhenMoving", type: "enum",
+				parameterNumber: 11, size: 1, defaultValue: 1,
+				values: [
+						0: "LED disabled",
+						1: "White",
+						2: "Red",
+						3: "Green",
+						4: "Blue",
+						5: "Yellow",
+						6: "Cyan",
+						7: "Magenta"
+				],
+				description: "This setting defines the LED colour when the motor is running."
+		],
+		[
+				name: "LED frame - colour when not moving", key: "ledFrame-ColourWhenNotMoving", type: "enum",
+				parameterNumber: 12, size: 1, defaultValue: 0,
+				values: [
+						0: "LED disabled",
+						1: "White",
+						2: "Red",
+						3: "Green",
+						4: "Blue",
+						5: "Yellow",
+						6: "Cyan",
+						7: "Magenta"
+				],
+				description: "This setting defines the LED colour when the motor isn't running."
+		],
+		[
+				name: "LED frame - brightness", key: "ledFrame-Brightness", type: "boolRange",
+				parameterNumber: 13, size: 1, defaultValue: 100,
+				range: "1..100", disableValue: 0,
+				description: "This setting allows to adjust the LED frame brightness."
+		],
+		[
+				name: "Operating mode", key: "operatingMode", type: "enum",
+				parameterNumber: 151, size: 1, defaultValue: 1,
+				values: [
+						1: "Roller blind (with positioning)",
+						2: "Venetian blind (with positioning)",
+						5: "Roller blind with built-in driver",
+						6: "Roller blind with built-in driver (impulse)"
+				],
+				description: "This setting allows adjusting operation according to the connected device."
+		],
+		[
+				name: "Delay motor stop after reaching end switch", key: "delayMotorStopAfterReachingEndSwitch", type: "range",
+				parameterNumber: 154, size: 2, defaultValue: 10,
+				range: "1..255",
+				description: "The setting determines the time after which the motor will be stopped after end switch contacts are closed."
+		],
+		[
+				name: "Motor operation detection", key: "motorOperationDetection", type: "range",
+				parameterNumber: 155, size: 2, defaultValue: 10,
+				range: "1..255",
+				description: "Power threshold interpreted as reaching a limit switch."
+		],
+		[
+				name: "Time of up movement", key: "timeOfUpMovement", type: "range",
+				parameterNumber: 156, size: 4, defaultValue: 6000,
+				range: "1..90000", 
+				description: "This setting determines the time needed for roller blinds to reach the top. [100 = 1s]"
+		],
+		[
+				name: "Time of down movement", key: "timeOfDownMovement", type: "range",
+				parameterNumber: 157, size: 4, defaultValue: 6000,
+				range: "1..90000", 
+				description: "This setting determines time needed for roller blinds to reach the bottom. [100 = 1s]" 
```

Comment:
so these are necessary to set if you want the device to run properly?

Reply:
Yes. Without those, user is left with default 1 min up and 1 min down times with no possibility to change.